libs/server/features/src/providers/teller/teller.service.ts

```diff
@@ -115,12 +126,67 @@ export class TellerService implements IAccountConnectionProvider, IInstitutionPr
             return {
                 providerId: id,
                 name,
-                url: undefined,
-                logo: `https://teller.io/images/banks/${id}.jpg}`,
-                primaryColor: undefined,
-                oauth: undefined,
+                url: null,
+                logo: null,
+                logoUrl: `https://teller.io/images/banks/${id}.jpg`,
+                primaryColor: null,
+                oauth: false,
                 data: tellerInstitution,
             }
         })
     }
+
+    async handleEnrollment(
+        userId: User['id'],
+        institution: Pick<TellerTypes.Institution, 'name' | 'id'>,
+        enrollment: TellerTypes.Enrollment
+    ) {
+        const connections = await this.prisma.accountConnection.findMany({
+            where: { userId },
+        })
+
+        if (connections.length > 40) {
+            throw new Error('MAX_ACCOUNT_CONNECTIONS')
+        }
+
+        const accounts = await this.teller.getAccounts({ accessToken: enrollment.accessToken })
+
+        this.logger.info(`Teller accounts retrieved for enrollment ${enrollment.enrollment.id}`)
+
+        // If all the accounts are Non-USD, throw an error
+        if (accounts.every((a) => a.currency !== 'USD')) {
+            throw new Error('USD_ONLY')
+        }
+
+        await this.prisma.user.update({
+            where: { id: userId },
+            data: {
+                tellerUserId: enrollment.user.id,
+            },
+        })
+
+        const accountConnection = await this.prisma.accountConnection.create({
+            data: {
+                name: enrollment.enrollment.institution.name,
+                type: 'teller' as SharedType.AccountConnectionType,
+                tellerEnrollmentId: enrollment.enrollment.id,
+                tellerInstitutionId: institution.id,
+                tellerAccessToken: this.crypto.encrypt(enrollment.accessToken),
+                userId,
+                syncStatus: 'PENDING',
+            },
+        })
+
+        await this.sync(accountConnection, { type: 'teller', initialSync: true })
```

Comment:
there's probably a better way to do this, need to study the syncing paths better and the worker services